### ISSUE TITLE
OpenEmu takes 100% CPU when PS4 DualShock connected

### ISSUE THREAD

**Issue by mrbodich** [OpenEmu/OpenEmu#4079]
Summary of issue

When I connect DualShock 4 with Bluetooth, OpenEmu start eating 100% CPU and almost freezes. When I disconnect DualShock, OpenEmu became normal. No relaunch needed. Just connect/disconnect DualShock.

OpenEmu 2.1 works fine by the way! So it's easier to find a problem now I guess.

- OpenEmu Version: 2.2
- macOS Version: 10.15.2

**Comment by clobber** [comment]
I use an official PS4 DualShock 4 connected via Bluetooth and cannot replicate this. Is that the same device as yours?

Please thoroughly test this build to see if it resolves the problem:
https://send.firefox.com/download/252626e07db00654/#MQKyyqlBcBZLEXljAqH_pA

**Comment by mrbodich** [comment]
Yes, I have DualShock 4 too (mistyped with 2, fixed)
And this attached build loads CPU by 100% too with it. 2.1 still works fine.

**Comment by mrbodich** [comment]
2.1 now loads CPU too((( It worked fine only once until OpenEmu relaunch

**Comment by clobber** [comment]
>2.1 now loads CPU too((( It worked fine only once until OpenEmu relaunch

Ok, maybe not an OpenEmu bug here, but try running the following command in Terminal.app and then paste the file it generates back here.

`sample $(pgrep OpenEmu) > ~/Desktop/oe_sample_results.txt`

**Comment by clobber** [comment]
Also a guide on how to do this from Activity Monitor: https://support.omnigroup.com/sample-osx/

**Comment by mrbodich** [comment]
I've launched on another user, and it worked fine there. Now works well on my user too. So for now it's good. I will try that method if I will fall into a trouble again. Thanks

**Comment by mrbodich** [comment]
Hi, open this thread again please. I have CPU issue again and again. Only sometimes it works fine... I did Sample from Activity Monitor, attaching it here.

[Sample «OpenEmu».txt](https://github.com/OpenEmu/OpenEmu/files/4023539/Sample.OpenEmu.txt)

**Comment by shysaur** [comment]
From your sampling results, it seems that your issue is caused by performance problems in macOS Catalina.

We think we have identified the problem; please download the OpenEmu build from the link below, and report if it solves your problem or not. (note: the link will expire in 7 days or after 20 downloads)

https://send.firefox.com/download/4ea6f60a7d3cdba9/#aXAiYA7S1m_pSrUW0t8U9A

**Comment by chadlavi** [comment]
I'm experiencing (what I think is) a similar issue to the original poster; I'm using v2.2 on macOS Catalina (10.15.3 beta). 

I'm trying to use a retrolink n64 controller, but it seems that _only_ the dpad and stick can be mapped in the controls settings. Start, AB, C-pad, L, R, and Z cannot be mapped at all, and pressing them has no effect.

**Comment by clobber** [comment]
@chadlavi There is a test build right above your post...

**Comment by chadlavi** [comment]
@clobber oh yikes -- I totally commented in the wrong issue thread here! I'll delete my comment and move it to the right place.

**Comment by mrbodich** [comment]
> From your sampling results, it seems that your issue is caused by performance problems in macOS Catalina.
> 
> We think we have identified the problem; please download the OpenEmu build from the link below, and report if it solves your problem or not. (note: the link will expire in 7 days or after 20 downloads)
> 
> https://send.firefox.com/download/4ea6f60a7d3cdba9/#aXAiYA7S1m_pSrUW0t8U9A

I’ve tested several times and did not have cpu problems yet with this version. So possible that it’s fixed. I can test more and tell you if any changes.

**Comment by clobber** [comment]
Thanks for testing and confirming.

Turns out this was a duplicate of #4044 so I'll close this one.

**Comment by mrbodich** [comment]
> Thanks for testing and confirming.
> 
> Turns out this was a duplicate of #4044 so I'll close this one.

Don't think that's the same trouble, I did not have lags after keys pressing. Anyway, the last build you've sent here worked well. Thank you for such fast fix!

**Comment by clobber** [comment]
>Don't think that's the same trouble, I did not have lags after keys pressing

Based on the sample output, it is indeed the same general problem caused by a bug in 10.15.
